File: Ksenia Kuzmina/Term 4/Task 6 (4.1)/PeerToPeerChat/Network/Listener.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.Net.Sockets;

namespace PeerToPeerChat.Network
{
    public class Listener
    {
        private Socket socket;
        public IPEndPoint Bound;

        public Listener(int port)
        {
            var localEndPoint = new IPEndPoint(IPAddress.Any, port);

            socket = new Socket(IPAddress.Any.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
            Bound = localEndPoint;

            socket.Bind(localEndPoint);
            socket.Listen(10);
        }

        public Connection? Accept()
        {
	        try
	        {
		        var res = socket.Accept();

		        return new Connection(res);
	        }
	        catch
	        {
		        return null;
	        }

        }

        public void Close()
        {
			socket.Close();
        }
    }
}

File: IgorKorobitsyn/Block4/Task6/src/main/kotlin/peerToPeerChat/Listener.kt
package peerToPeerChat
import java.net.InetAddress
import java.net.InetSocketAddress
import java.net.ServerSocket

class Listener(port: Int) {
    private val serverSocket: ServerSocket = ServerSocket()
    private val bound: InetSocketAddress

    init {
        val localEndPoint = InetSocketAddress(InetAddress.getByName(null), port)

        serverSocket.bind(localEndPoint)
        bound = serverSocket.localSocketAddress as InetSocketAddress
    }

    fun accept(): Connection? {
        return try {
            val clientSocket = serverSocket.accept()
            return Connection(clientSocket)
        } catch (e: Exception) {
            println(e.message)
            println("In Listener.accept")
            null
        }
    }

    fun close() {
        serverSocket.close()
    }
}
